frontend/public/components/storage/create-pvc.tsx

```diff
@@ -173,6 +165,43 @@ export const CreatePVCForm: React.FC<CreatePVCFormProps> = (props) => {
 
   const onlyPvcSCs = React.useCallback((sc: StorageClass) => !isObjectSC(sc), []);
 
+  const getVolumeMode = () => {
+    if (storageProvisioner.includes(cephStorageProvisioners[1])) {
+      return [
+        <>
+          Filesystem{' '}
+          <InfoToolTip
+            position="bottom"
+            content={<div>Only filesystem volume mode is available for cephfs</div>}
+          />
+        </>,
+      ];
+    }
+
+    if (storageProvisioner.includes(cephStorageProvisioners[2]) && accessMode === 'ReadWriteMany') {
+      return [
+        <>
+          Block{' '}
+          <InfoToolTip
+            position="bottom"
+            content={<div>Only block volume mode is available for rbd with ReadWriteMany mode</div>}
+          />
+        </>,
+      ];
+    }
+
+    return volumeModeRadios.map((radio) => (
+      <RadioInput
+        {...radio}
+        key={radio.value}
+        onChange={handleVolumeMode}
+        inline
+        checked={radio.value === volumeMode}
+        name="volumeMode"
+      />
+    ));
+  };
```

Comment:
Looks a bit hacky to me.

Let do it in a proper way so that moving forward any other provisioner can also leverage this functionality if they don't support any volume mode.

Let define an object with the keys pointing to the provisioner name and values being an object that says volume-mode.

Reply:
i don't understand this approach.... Could you elaborate ? 
We have only two cases to look out for here... 
When provisioner is cephfs and any access mode ; we want the volume mode to always be 'Filesystem' 
When provisioner is ceph rbd and only RWX access mode; we want the volume mode to always be 'Block'

Having an map defined for only two very specific conditions seems to be convoluted approach (as per what i understand). My suggestion is to move this function to shared if we want re-usability . We can change to different approach once more conditions come up.  WDYT ?